data/data/aws/vpc/common.tf

```diff
@@ -17,10 +17,11 @@ locals {
   vpc_id = "${aws_vpc.new_vpc.id}"
 
   // When referencing the _ids arrays or data source arrays via count = , always use the *_count variable rather than taking the length of the list
-  private_subnet_ids   = "${aws_subnet.private_subnet.*.id}"
-  public_subnet_ids    = "${aws_subnet.public_subnet.*.id}"
-  private_subnet_count = "${local.new_az_count}"
-  public_subnet_count  = "${local.new_az_count}"
+  private_subnet_ids      = "${aws_subnet.private_subnet.*.id}"
+  public_subnet_ids       = "${aws_subnet.public_subnet.*.id}"
+  private_subnet_count    = "${local.new_az_count}"
+  public_subnet_count     = "${local.new_az_count}"
+  az_to_private_subnet_id = "${zipmap(local.new_subnet_azs, local.private_subnet_ids)}"
```

Comment:
nit: This seems to be used only in output, why not do the calculation there itself ...

Reply:
^^ totaly ignorable :innocent: